tests/farmer_harvester/test_third_party_harvesters.py

```diff
@@ -0,0 +1,441 @@
+        FarmerService,
+        HarvesterService,
+        Union[FullNodeService, SimulatorFullNodeService],
+        Union[FullNodeService, SimulatorFullNodeService],
+        BlockTools,
+    ],
+    caplog: pytest.LogCaptureFixture,
+    mocker: MockerFixture,
+) -> None:
+    """
+    Tests fee convention specified in CHIP-22: https://github.com/Chia-Network/chips/pull/88
+    """
+    (
+        farmer_service,
+        _,
+        full_node_service_1,
+        full_node_service_2,
+        _,
+    ) = farmer_harvester_2_simulators_zero_bits_plot_filter
+
+    farmer: Farmer = farmer_service._node
+    full_node_1: FullNode = full_node_service_1._node
+    full_node_2: FullNode = full_node_service_2._node
+
+    # Connect peers to each other
+    farmer_service.add_peer(
+        UnresolvedPeerInfo(str(full_node_service_2.self_hostname), full_node_service_2._server.get_port())
+    )
+    full_node_service_2.add_peer(
+        UnresolvedPeerInfo(str(full_node_service_1.self_hostname), full_node_service_1._server.get_port())
+    )
+
+    await wait_until_node_type_connected(farmer.server, NodeType.FULL_NODE)
+    await wait_until_node_type_connected(farmer.server, NodeType.HARVESTER)  # Should already be connected
+    await wait_until_node_type_connected(full_node_1.server, NodeType.FULL_NODE)
+
+    fee_threshold = 0.5
+    max_fee_proofs = 5
+    fee_count = 0
+    proof_count = 0
+
+    farmer_reward_puzzle_hash = decode_puzzle_hash("txch1psqeaw0h244v5sy2r4se8pheyl62n8778zl6t5e7dep0xch9xfkqhx2mej")
+
+    async def intercept_farmer_new_proof_of_space(*args: Any) -> None:
+        nonlocal farmer
+        nonlocal fee_threshold
+        nonlocal max_fee_proofs
+        nonlocal proof_count
+        nonlocal fee_count
+        nonlocal farmer_reward_puzzle_hash
+
+        request: harvester_protocol.NewProofOfSpace = harvester_protocol.NewProofOfSpace.from_bytes(args[0])
+
+        fee_threshold_int = uint32(int(0xFFFFFFFF * fee_threshold))
+
+        fee_quality = calculate_harvester_fee_quality(request.proof.proof, request.challenge_hash)
+        if fee_quality <= fee_threshold_int and fee_count < max_fee_proofs:
+            fee_count += 1
+            request = dataclasses.replace(
+                request,
+                farmer_reward_address_override=farmer_reward_puzzle_hash,
+                fee_info=ProofOfSpaceFeeInfo(applied_fee_threshold=fee_threshold_int),
+            )
+
+        if proof_count <= max_fee_proofs:
+            proof_count += 1
+
+        peer: WSChiaConnection = args[1]
+        await FarmerAPI.new_proof_of_space(farmer.server.api, request, peer)
+
+    mocker.patch.object(farmer.server.api, "new_proof_of_space", side_effect=intercept_farmer_new_proof_of_space)
+
+    log_text_len = 0
+
+    def log_has_new_text() -> bool:
+        nonlocal caplog
+        nonlocal log_text_len
+
+        text_len = len(caplog.text)
+        if text_len > log_text_len:
```

Comment:
this is always true, in the test. I take it that's expected. Maybe you could mark the else-case as `no-cover`